engines/e_afalg.c

```diff
@@ -355,13 +355,26 @@ static int afalg_fin_cipher_aio(afalg_aio *aio, int sfd, unsigned char *buf,
                         }
                         continue;
                     } else {
+                        char strbuf[32];
+
                         /*
                          * Retries exceed for -EBUSY or unrecoverable error
                          * condition for this instance of operation.
                          */
                         ALG_WARN
                             ("%s(%d): Crypto Operation failed with code %lld\n",
                              __FILE__, __LINE__, events[0].res);
+                        BIO_snprintf(strbuf, sizeof(strbuf), "%lld", events[0].res);
+                        switch (events[0].res) {
+                        case -ENOMEM:
+                            AFALGerr(AFALG_F_KERNEL_OP, AFALG_R_KERNEL_OP_FAILED);
+                            ERR_add_error_data(3, "-ENOMEM ( code ", strbuf, " )");
+                            break;
+                        default:
+                            AFALGerr(AFALG_F_KERNEL_OP, AFALG_R_KERNEL_OP_FAILED);
```

Comment:
!

Reply:
i added AFALG_F_KERNEL_OP to e_afalg_err.c / e_afalg_err.h

Reply:
It's not quite that easy.  The name should match the name of this function in that case, so it should be `AFALG_F_AFALG_FIN_CIPHER_AIO`.  But, these function codes are *irrelevant* since OpenSSL 3.0, so it's much easier to simply have `0` there.

Reply:
ah ok i'll changed that